generators/build/app.js

```diff
@@ -164,9 +166,12 @@ function getBuild(args, callback) {
       });
    } else {
       util.isTFSGreaterThan2017(args.tfs, pat, (e, result) => {
-         if (result) {
+         if (kubeDeployment) {
+            build = `vsts_kube_${kubeDeployment}_build.json`;
```

Comment:
It seems like ${args.type} is the parameter and you need it set to kube_${kubeDeployment}. Try to abstract that out oppose to make these if/else cases.